Source/Core/VideoBackends/Metal/MTLMain.mm

```diff
@@ -96,8 +96,11 @@ static bool WindowSystemTypeSupportsMetal(WindowSystemType type)
     Util::PopulateBackendInfoFeatures(&g_Config, adapter);
 
 #if TARGET_OS_OSX
-    if (@available(macOS 13.3, *))
-      [adapter setShouldMaximizeConcurrentCompilation:YES];
+    if (@available(macOS 13.3, *)) {
+      if ([adapter respondsToSelector:@selector(setShouldMaximizeConcurrentCompilation:)]) {
```

Comment:
We need the double-if there, as it seems `if(@available(…))` can only take a single condition.

Reply:
Personally I'd prefer if we replaced the `@available` and disabled the associated warning with [one of these](https://github.com/dolphin-emu/dolphin/blob/937cb8ef3a0a413e9c76c0cf90cffd8a935ab595/Source/Core/VideoBackends/Metal/MTLUtil.mm#L284-L312), as the `@available` is useless if it doesn't actually work.

Also add a comment mentioning that the 13.3 check doesn't work even though the headers say it should, mentioning the exact cpu model and macOS version it fails on (get it with `sysctl machdep.cpu.brand_string`)

Reply:
Done (CPU model, comment, and the availability check).